sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -315,16 +338,29 @@ public BlobBaseClient(Uri blobUri, TokenCredential credential, BlobClientOptions
         /// policies for authentication, retries, etc., that are applied to
         /// every request.
         /// </param>
-        internal BlobBaseClient(Uri blobUri, HttpPipelinePolicy authentication, BlobClientOptions options)
+        /// <param name="storageSharedKeyCredential">
+        /// The shared key credential used to sign requests.
+        /// </param>
+        internal BlobBaseClient(
+            Uri blobUri,
+            HttpPipelinePolicy authentication,
+            BlobClientOptions options,
+            StorageSharedKeyCredential storageSharedKeyCredential)
         {
             options ??= new BlobClientOptions();
             _uri = blobUri;
+            if (!string.IsNullOrEmpty(blobUri.Query))
+            {
+                _snapshot = System.Web.HttpUtility.ParseQueryString(blobUri.Query).Get(Constants.SnapshotParameterName);
+                _blobVersionId = System.Web.HttpUtility.ParseQueryString(blobUri.Query).Get(Constants.VersionIdParameterName);
```

Comment:
Should we be using https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Common/src/Shared/UriQueryParamsCollection.cs instead?  And parsing is kind of expensive so it'd be nice to do that once and save the value in a variable to use for each lookup.